source/models/index.py

```diff
@@ -0,0 +1,45 @@
+import os
+# from models.table import Table
+from outputs.exceptions import *
+
+class Index:
+    def __init__(self, table, index_name):
+        self.table = table
+        self.index_name = index_name
+
+    def serialize(self):
+        os.makedirs(self.get_path(), exist_ok = True)
+    
+    def get_path(self):
+        return os.path.join(self.table.get_path(), "indices\\" + self.index_name)
+    
+    def add_primary_key(self, primary_key, index_value):
+        index_value_file_path = os.path.join(self.get_path(), index_value + ".txt")
+        old_primary_keys = set()
```

Comment:
you can have a public method call `get_primary_keys` and call it in everywhere

Reply:
it returns empty array if the file doesn't exist

Reply:
here

Reply:
why we need it to be a public are we will need it in other class ?

Reply:
```
primary_keys = get_primary_keys
primary_keys.add(primary_key)
__update_primary_keys(primary_keys)

```